templates/worker/01-worker-kubelet/_base/units/kubelet.service.yaml

```diff
@@ -5,6 +5,7 @@ contents: |
   Description=Kubernetes Kubelet
   Wants=rpc-statd.service network-online.target crio.service
   After=network-online.target crio.service
+  After=ostree-finalize-staged.service
```

Comment:
Is it worth putting a condition in the template so that this is only used for RHCOS nodes?

Reply:
good point, this will work on OSTree based nodes but break RHEL nodes

Reply:
That's a good idea!

Though hmm... AFAICT there doesn't seem to be a variable already exposed for this. So I think this would require wiring it through the render code first. Meh... maybe not worth the trouble in that case? (It's totally harmless on non-RHCOS nodes.)

Reply:
> good point, this will work on OSTree based nodes but break RHEL nodes

systemd just ignores `Before=`/`After=` constraints if the targetunit is missing.

Reply:
cool, RHEL nodes are safe then :laughing: